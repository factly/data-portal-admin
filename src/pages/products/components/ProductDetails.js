import React from 'react';
import { Descriptions, Card, Tag } from 'antd';
import moment from 'moment';

const ProductDetails = ({ product, currencyCode, productTags }) => {
  return (
    <Card>
      <Descriptions title={product.title} column={1}>
        <Descriptions.Item label="Slug">{product.slug}</Descriptions.Item>
        <Descriptions.Item label="Price">
          {product.price} {currencyCode}
        </Descriptions.Item>
        <Descriptions.Item label="Status">{product.status}</Descriptions.Item>
        <Descriptions.Item label="Tags">
          {productTags.map((tag) => (
            <Tag key={tag.id}>{tag.title}</Tag>
          ))}
        </Descriptions.Item>
        <Descriptions.Item label="Created At">
          {moment(product.created_at).fromNow()}
        </Descriptions.Item>
      </Descriptions>
    </Card>
  );
};

export default ProductDetails;
