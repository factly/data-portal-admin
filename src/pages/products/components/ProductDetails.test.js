import React from 'react';
import { shallow } from 'enzyme';

import ProductDetails from './ProductDetails';

describe('ProductDetails component', () => {
  const product = {
    id: 1,
    title: 'title',
    slug: 'slug',
    price: 'price',
    status: 'status',
    created_at: '2020-12-12',
  };

  it('should render the component', () => {
    const component = shallow(
      <ProductDetails
        product={product}
        currencyCode={'INR'}
        productTags={[{ id: 1, title: 'tag1' }]}
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
