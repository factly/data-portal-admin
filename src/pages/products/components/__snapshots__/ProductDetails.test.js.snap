// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetails component should render the component 1`] = `
<Card>
  <Descriptions
    column={1}
    title="title"
  >
    <DescriptionsItem
      label="Slug"
    >
      slug
    </DescriptionsItem>
    <DescriptionsItem
      label="Price"
    >
      price
       
      INR
    </DescriptionsItem>
    <DescriptionsItem
      label="Status"
    >
      status
    </DescriptionsItem>
    <DescriptionsItem
      label="Tags"
    >
      <Tag
        key="1"
      >
        tag1
      </Tag>
    </DescriptionsItem>
    <DescriptionsItem
      label="Created At"
    >
      4 months ago
    </DescriptionsItem>
  </Descriptions>
</Card>
`;
